File: twoSum.c
// 1. 两数之和
// https://leetcode-cn.com/problems/two-sum/

/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* nums, int numsSize, int target, int* returnSize){
    for(int i=0; i < numsSize; i++){
        int tmp = target - nums[i];
        for(int j=i+1; j < numsSize; j++){
            if(nums[j] != tmp) continue;
            *returnSize = 2;
            int* arr = (int*)malloc(sizeof(int)*2);
            arr[0] = i;
            arr[1] = j;
            return arr;
        }
    }
    *returnSize = 0;
    return NULL;
}

File: two_sum.rs
// 1. 两数之和
// https://leetcode-cn.com/problems/two-sum/

use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut map= HashMap::new();
        for (i,v) in nums.iter().enumerate() {
            let sub = target - v;
            if map.contains_key(v) {
                let j = map[v];
                return vec![j as i32, i as i32];
            }else{
                map.insert(sub, i);
            }
        }
        return vec![0, 0];
    }
    
}
